File: iTransform-experimental/src/app/user/user.service.ts
import { Injectable } from "@angular/core";
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Courses } from '../course/course';
import { User, Authenticate } from './user';


@Injectable()
export class UserService{

    baseUrl = "http://localhost:8083/user";
    baseUrlForCourse = "http://localhost:8084/course"

    constructor(private http:HttpClient){}

    getAllCourses():Observable<Courses[]>{
        return this.http.get<Courses[]>(this.baseUrlForCourse);
        }
        
    addNewUser(user:User):Observable<User>{
        return this.http.post<User>(this.baseUrl,user);
    }

    getUserAuthentication(userName,password):Observable<User>{
        return this.http.get<User>(this.baseUrl+"/" + userName+ "/"+password);
    }
}

File: user-service/src/main/java/com/capgemini/itransform/userservice/service/UserService.java
package com.capgemini.itransform.userservice.service;

import com.capgemini.itransform.userservice.entity.User;

public interface UserService {

	public User addNewUser(User user);
	
	public User getUserByUserName(String userName,String password);
	
	public User updateUser(User user);
}
